orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -163,28 +163,28 @@ public TaskResult execute(@Nonnull Stage stage) {
             cluster.get("serverGroups"), new TypeReference<List<ServerGroup>>() {});
     serverGroups.sort(Comparator.comparing((ServerGroup o) -> o.createdTime).reversed());
 
+    List<ServerGroup> enabledServerGroups =
+        serverGroups.stream()
+            .filter(s -> s.disabled == null || !s.disabled)
+            .collect(Collectors.toList());
+
     List<Map> imagesToRestore = new ArrayList<>();
     for (String region : stageData.regions) {
-      List<ServerGroup> serverGroupsInRegion =
-          serverGroups.stream()
+      List<ServerGroup> enabledServerGroupsInRegion =
+          enabledServerGroups.stream()
```

Comment:
I think you could also filter for not null here, and keep the rename to  `enabledServerGroupsInRegion`. It doesn't seem like you use the `enabledServerGroups` list anywhere else.

Reply:
I'm not sure I follow... Filter on not-null on which property, exactly?

Reply:
Ah sorry, filter out the disabled server groups!

Reply:
This is a different list, though -- the filter here is to take only the enabled server groups in one region.

Reply:
never mind!